r/R/parquet.R

```diff
@@ -34,9 +121,11 @@
 #' }
 #'
 #' @export
-read_parquet <- function(file, as_tibble = TRUE, ...) {
-  tab <- shared_ptr(`arrow::Table`, read_parquet_file(file))
-  if (isTRUE(as_tibble)) {
+read_parquet <- function(file, props = parquet_arrow_reader_properties(), as_tibble = TRUE, col_select = NULL, ...) {
+  reader <- parquet_file_reader(file, props = props, ...)
+  tab <- reader$ReadTable(!!enquo(col_select))
```

Comment:
Why does this reader have a `ReadTable` method while the others just have `Read`?

Reply:
That's the name of the method in the C++ api:

https://github.com/apache/arrow/blob/master/cpp/src/parquet/arrow/reader.h#L217